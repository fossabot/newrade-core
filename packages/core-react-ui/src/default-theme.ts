/**
 *
 * Export the default theme values
 *
 */

export * from './default-theme/default-animations';
export * from './default-theme/default-buttons';
export * from './default-theme/default-code-colors';
export * from './default-theme/default-code-colors-dark';
export * from './default-theme/default-colors';
export * from './default-theme/default-colors-dark';
export * from './default-theme/default-effects';
export * from './default-theme/default-fonts';
export * from './default-theme/default-iconography';
export * from './default-theme/default-layout';
export * from './default-theme/default-sizing';
export * from './default-theme/default-typography';
