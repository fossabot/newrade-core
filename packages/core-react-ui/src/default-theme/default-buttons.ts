import { BorderStyles, Buttons } from '@newrade/core-design-system';

import { defaultColorIntents, defaultColors } from './default-colors';

export const defaultButtonBorder: BorderStyles = {
  default: {
    width: 1,
    style: 'solid',
    color: defaultColors.colorIntents.primary,
    radius: {
      all: 3,
    },
  },
  top: {},
  right: {},
  bottom: {},
  left: {},
};

export const defaultButtons: Buttons = {
  sizes: {
    large: {
      padding: {
        default: NaN,
        left: 16,
        right: 16,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 12,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 4,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      iconSize: 24,
      iconMargin: {
        left: 10,
        right: 10,
        top: NaN,
        bottom: NaN,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
    medium: {
      padding: {
        default: NaN,
        left: 10,
        right: 10,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 10,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 4,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
    small: {
      padding: {
        default: NaN,
        left: 8,
        right: 8,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 8,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 3,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
    xSmall: {
      padding: {
        default: NaN,
        left: 6,
        right: 6,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 6,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 3,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
  },
  variants: {
    primary: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    secondary: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    tertiary: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    positive: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    warning: {
      textColor: defaultColorIntents.warningText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    danger: {
      textColor: defaultColorIntents.dangerText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
  },
};

/**
 * DefaultButtons object without the `vars` property which is used only
 * in application-defined theme configuration
 */
export type DefaultButtons = Omit<Buttons, 'vars'>;
