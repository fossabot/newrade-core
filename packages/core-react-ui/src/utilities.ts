export * from './utilities/colors.utilities';
export * from './utilities/component.utilities';
export * from './utilities/css-variable.utilities';
export * from './utilities/effects.utilities';
export * from './utilities/filter-color.utility';
export * from './utilities/log.utilities';
export * from './utilities/text-capsize.utilities';
export * from './utilities/text.utilities';
export * from './utilities/theme.utilities';
export * from './utilities/utilities';

