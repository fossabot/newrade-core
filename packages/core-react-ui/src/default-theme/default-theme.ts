import { COLOR_SCHEME, DesignSystemV2 } from '@newrade/core-design-system';

import { defaultAnimations } from './default-animations';
import { defaultButtons } from './default-buttons';
import { defaultColors } from './default-colors';
import { defaultEffects } from './default-effects';
import { defaultIconography } from './default-iconography';
import { defaultLayout } from './default-layout';
import { defaultSizing } from './default-sizing';
import { defaultTypography } from './default-typography';

export const defaultTheme: DesignSystemV2 = {
  name: 'core-react-ui',
  colorScheme: COLOR_SCHEME.LIGHT,
  colors: defaultColors,
  effects: defaultEffects,
  sizing: defaultSizing,
  iconography: defaultIconography,
  typography: defaultTypography,
  layout: defaultLayout,
  animations: defaultAnimations,
  components: {
    buttons: defaultButtons,
  },
};
