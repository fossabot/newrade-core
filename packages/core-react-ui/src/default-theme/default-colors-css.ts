import { getCSSColorsV2 } from '../utilities-theme/colors.utilities';

import { defaultColors } from './default-colors';

export const defaultCSSColors = getCSSColorsV2(defaultColors);
