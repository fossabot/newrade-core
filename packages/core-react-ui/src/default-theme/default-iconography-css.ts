import { defaultIconography } from '../default-theme';
import { getCSSIconography } from '../utilities-theme/iconography.utilities';

export const defaultCSSIconography = getCSSIconography(defaultIconography);
