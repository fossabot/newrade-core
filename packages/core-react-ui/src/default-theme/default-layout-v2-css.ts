import { defaultLayout } from '../default-theme';
import { getCSSLayoutV2 } from '../utilities-theme/layout.utilities';

export const defaultCSSLayoutV2 = getCSSLayoutV2(defaultLayout);
