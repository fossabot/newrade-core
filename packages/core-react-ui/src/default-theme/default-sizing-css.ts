import { defaultSizing } from '../default-theme';
import { getCSSSizingV2 } from '../utilities-theme/sizing.utilities';

export const defaultCSSSizing = getCSSSizingV2(defaultSizing);
