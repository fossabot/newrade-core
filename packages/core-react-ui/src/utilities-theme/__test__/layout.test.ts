import { Layout } from '@newrade/core-design-system';

import { defaultLayout } from '../../default-theme/default-layout';
import { getCSSLayout } from '../layout.utilities';

describe('layout utilities', () => {
  describe(`${getCSSLayout.name}`, () => {
    it('should create a valid CSS layout config', () => {
      const layout = getCSSLayout(defaultLayout) as Layout<string>;
      expect(layout).toMatchSnapshot();
    });
  });
});
