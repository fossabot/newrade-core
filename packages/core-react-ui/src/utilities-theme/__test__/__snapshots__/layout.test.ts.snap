// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout utilities getCSSLayout should create a valid CSS layout config 1`] = `
Object {
  "asideWidth": "260px",
  "breakpoints": Object {
    "desktopLarge": "1600px",
    "desktopMedium": "1440px",
    "desktopSmall": "1280px",
    "desktopXLarge": "1920px",
    "mobileMedium": "414px",
    "mobileSmall": "375px",
    "mobileXSmall": "320px",
    "tabletLandscape": "1024px",
    "tabletPortrait": "768px",
  },
  "contentMargins": Object {
    "desktop": "40px",
    "mobile": "24px",
    "tablet": "40px",
  },
  "contentWidth": Object {
    "desktopBlogMaxWidth": "800px",
    "desktopDocsMaxWidth": "800px",
    "desktopMaxWidth": "1200px",
  },
  "footerHeight": Object {
    "desktop": "210px",
    "mobile": "210px",
    "tablet": "210px",
  },
  "media": Object {
    "desktopLarge": "screen and (min-width: 1600px)",
    "desktopLargeOnly": "screen and (min-width: 1600px) and (max-width: 1920px)",
    "desktopMedium": "screen and (min-width: 1440px)",
    "desktopMediumOnly": "screen and (min-width: 1440px) and (max-width: 1600px)",
    "desktopSmall": "screen and (min-width: 1280px)",
    "desktopSmallOnly": "screen and (min-width: 1280px) and (max-width: 1440px)",
    "desktopXLarge": "screen and (min-width: 1920px)",
    "desktopXLargeOnly": "screen and (min-width: 1920px)",
    "mobile": "screen and (max-width: 768px)",
    "mobileOnly": "screen and (min-width: 0px) and (max-width: 768px)",
    "tablet": "screen and (min-width: 768px)",
    "tabletOnly": "screen and (min-width: 768px) and (max-width: 1280px)",
  },
  "navbarHeight": Object {
    "desktop": "64px",
    "mobile": "56px",
    "tablet": "56px",
  },
  "sidebarWidth": Object {
    "desktop": "270px",
    "mobile": "375px",
    "tablet": "270px",
  },
  "var": Object {
    "asideWidth": "var(--layout-aside-width)",
    "breakpoints": Object {
      "desktopLarge": "var(--layout-breakpoints-desktop-large)",
      "desktopMedium": "var(--layout-breakpoints-desktop-medium)",
      "desktopSmall": "var(--layout-breakpoints-desktop-small)",
      "desktopXLarge": "var(--layout-breakpoints-desktop-x-large)",
      "mobileMedium": "var(--layout-breakpoints-mobile-medium)",
      "mobileSmall": "var(--layout-breakpoints-mobile-small)",
      "mobileXSmall": "var(--layout-breakpoints-mobile-x-small)",
      "tabletLandscape": "var(--layout-breakpoints-tablet-landscape)",
      "tabletPortrait": "var(--layout-breakpoints-tablet-portrait)",
    },
    "contentMargins": "var(--layout-content-margins)",
    "contentWidth": Object {
      "desktopBlogMaxWidth": "var(--layout-content-width-desktop-blog-max-width)",
      "desktopDocsMaxWidth": "var(--layout-content-width-desktop-docs-max-width)",
      "desktopMaxWidth": "var(--layout-content-width-desktop-max-width)",
    },
    "footerHeight": "var(--layout-footer-height)",
    "navbarHeight": "var(--layout-navbar-height)",
    "sidebarWidth": "var(--layout-sidebar-width)",
  },
  "varNames": Object {
    "asideWidth": "--layout-aside-width",
    "breakpoints": Object {
      "desktopLarge": "--layout-breakpoints-desktop-large",
      "desktopMedium": "--layout-breakpoints-desktop-medium",
      "desktopSmall": "--layout-breakpoints-desktop-small",
      "desktopXLarge": "--layout-breakpoints-desktop-x-large",
      "mobileMedium": "--layout-breakpoints-mobile-medium",
      "mobileSmall": "--layout-breakpoints-mobile-small",
      "mobileXSmall": "--layout-breakpoints-mobile-x-small",
      "tabletLandscape": "--layout-breakpoints-tablet-landscape",
      "tabletPortrait": "--layout-breakpoints-tablet-portrait",
    },
    "contentMargins": "--layout-content-margins",
    "contentWidth": Object {
      "desktopBlogMaxWidth": "--layout-content-width-desktop-blog-max-width",
      "desktopDocsMaxWidth": "--layout-content-width-desktop-docs-max-width",
      "desktopMaxWidth": "--layout-content-width-desktop-max-width",
    },
    "footerHeight": "--layout-footer-height",
    "navbarHeight": "--layout-navbar-height",
    "sidebarWidth": "--layout-sidebar-width",
  },
  "zIndex": Object {
    "chatBubble": 130,
    "content": 50,
    "dialog": 70,
    "navBar": 100,
    "notifications": 120,
    "sideBarDesktop": 90,
    "sideBarMobile": 110,
  },
}
`;
